lib/internal/repl.js

```diff
@@ -86,15 +103,25 @@ function setupHistory(repl, historyPath, ready) {
     }
 
     if (data) {
+      repl.history = data.split(/[\n\r]+/).slice(-repl.historySize);
+    } else if (oldHistoryPath) {
+      // Grab data from the older pre-v3.0 JSON NODE_REPL_HISTORY_FILE format.
+      repl._writeToOutput(
+          '\nConverting old JSON repl history to line-separated history.\n' +
+          `The new repl history file can be found at ${historyPath}.\n`);
+      repl._refreshLine();
```

Comment:
Maybe move that message to a point after the conversation and point users to unsetting the old env variable so it won't convert again on next run.